src/models/analytics-panel.ts

```diff
@@ -3,9 +3,74 @@ import Joi from 'joi';
 import { schema as analyticsQuerySchema, AnalyticsQuery } from './analytics-query';
 import { schema as analyticsVisualisationSchema, AnalyticsVisualisation } from './analytics-visualisation';
 
+const columnsPerResolution = {
+  lg: 16,
+  md: 12,
+  sm: 8,
+  xs: 4,
+};
+
+const layoutItemSchema = (columnsCount: number) => Joi.array().items(Joi.object().keys({
+  x: Joi.number().integer().max(columnsCount - 1).required(),
+  y: Joi.number().integer().required(),
+  width: Joi.number().integer().min(4).max(columnsCount),
+  height: Joi.number().integer().min(6),
+}))
+  .custom((layout: CardPosition[], helper) => {
```

Comment:
I struggle a bit with adding this kind of complexity to the sdk (like we also have at [BaseFieldConfiguration](https://github.com/withthegrid/platform-sdk/blob/0ebbe8fcdfa8af013ec36282e475022a79c20fec/src/models/fields/base-field-configuration.ts#L26) for example), but don't have a better alternative at the moment. Therefore let's go for this and if we see an opportunity to simplify, we can do that later.